src/snowflake/snowpark/session.py

```diff
@@ -558,11 +560,17 @@ def __init__(
                 _PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING, True
             )
         )
-        self._cte_optimization_enabled: bool = (
+        cte_optimization_enabled_version = (
             self._conn._get_client_side_session_parameter(
-                _PYTHON_SNOWPARK_USE_CTE_OPTIMIZATION_STRING, False
+                _PYTHON_SNOWPARK_USE_CTE_OPTIMIZATION_VERSION, ""
             )
         )
+        self._cte_optimization_enabled: bool = (
+            isinstance(cte_optimization_enabled_version, str)
```

Comment:
@sfc-gh-jdu it seems we will have couple of parameters to perform similar logic, how about extract a common utility function,
 is_feature_enabled_for_version(feature_version_parameter), the function will get the version configured for the feature, and check if the current version >= the configured version

Reply:
+1

Reply:
good point! Added such a function